File: src/main/java/bad/robot/radiate/teamcity/BuildLocatorBuilder.java
package bad.robot.radiate.teamcity;

@Deprecated
class BuildLocatorBuilder {

    private final StringBuilder locator = new StringBuilder();

    public static BuildLocatorBuilder latest(BuildType buildType) {
        return new BuildLocatorBuilder().with(buildType);
    }

    public static BuildLocatorBuilder running(BuildType buildType) {
        return new BuildLocatorBuilder().with(buildType).running();
    }

    BuildLocatorBuilder with(BuildType type) {
        withSeparator().append("buildType:").append(type.getId());
        return this;
    }

    BuildLocatorBuilder running() {
        withSeparator().append("running:true");
        return this;
    }

    private StringBuilder withSeparator() {
        if (locator.length() > 0)
            locator.append(",");
        return locator;
    }

    String build() {
        return locator.toString();
    }
}

File: src/main/scala/bad/robot/radiate/teamcity/BuildLocatorBuilder.scala
package bad.robot.radiate.teamcity

object BuildLocatorBuilder {
  
  def latest(buildType: BuildType): BuildLocatorBuilder = {
    new BuildLocatorBuilder().`with`(buildType)
  }

  def running(buildType: BuildType): BuildLocatorBuilder = {
    new BuildLocatorBuilder().`with`(buildType).running
  }
}

class BuildLocatorBuilder {
  private val locator = new StringBuilder

  private[teamcity] def `with`(`type`: BuildType): BuildLocatorBuilder = {
    withSeparator.append("buildType:").append(`type`.id)
    this
  }

  private[teamcity] def running: BuildLocatorBuilder = {
    withSeparator.append("running:true")
    this
  }

  private def withSeparator: StringBuilder = {
    if (locator.nonEmpty) locator.append(",")
    locator
  }

  private[teamcity] def build: String = locator.toString()
}